trollmoves/dispatcher.py

```diff
@@ -336,13 +374,15 @@ def create_dest_url(self, msg, client, item):
         parts = urlsplit(host)
         host_path = urlunsplit((parts.scheme, parts.netloc, path,
                                 parts.query, parts.fragment))
-        return host_path, connection_parameters
+        return host_path, connection_parameters, client
 
     def close(self):
         """Shutdown the dispatcher."""
         logger.info('Terminating dispatcher.')
         self.loop = False
         self.listener.stop()
+        if self.publisher:
+            self.publisher.stop()
         self.config_handler.close()
```

Comment:
Can we put each of these in  a try/except to make sure the following are executed too ?

Reply:
Maybe with a logging message giving the reason of the failure.

Reply:
Will do.